src/rust/aggregator/src/data/leaderboards.rs

```diff
@@ -0,0 +1,293 @@
+use anyhow::anyhow;
+use bigdecimal::BigDecimal;
+use chrono::{DateTime, Duration, Utc};
+use sqlx::{PgConnection, PgPool, Postgres, Transaction};
+
+use super::{Data, DataAggregationError, DataAggregationResult};
+
+pub struct Leaderboards {
+    pool: PgPool,
+    last_indexed_timestamp: Option<DateTime<Utc>>,
+}
+
+impl Leaderboards {
+    pub fn new(pool: PgPool) -> Self {
+        Self {
+            pool,
+            last_indexed_timestamp: None,
+        }
+    }
+}
+
+struct Competition {
+    id: i32,
+    start: DateTime<Utc>,
+    end: DateTime<Utc>,
+    prize: i64,
+    market_id: BigDecimal,
+    integrators_required: Vec<String>,
+}
+
+#[async_trait::async_trait]
+impl Data for Leaderboards {
+    fn model_name(&self) -> &'static str {
+        "Leaderboard"
+    }
+
+    fn ready(&self) -> bool {
+        self.last_indexed_timestamp.is_none()
+            || self.last_indexed_timestamp.unwrap() + Duration::seconds(5) < Utc::now()
+    }
+
+    async fn process_and_save_historical_data(&mut self) -> DataAggregationResult {
+        self.process_and_save_internal().await
+    }
+
+    fn poll_interval(&self) -> Option<std::time::Duration> {
```

Comment:
Why is this returning an option? Does no interval mean "update as often as possible"? (which doesn't seem like something we would ever want)

Reply:
This is a trait function. It returns an option because maybe other types of data do not have poll intervals.

Reply:
When no polling interval is present, a default (aggregator global) interval is used.